api/api-public-tenant-service-account.go

```diff
@@ -195,6 +209,103 @@ func (s *server) UpdateServiceAccount(ctx context.Context, in *pb.UpdateServiceA
 	}, nil
 }
 
+func (s *server) EnableServiceAccount(ctx context.Context, in *pb.ServiceAccountActionRequest) (res *pb.ServiceAccount, err error) {
+	idRequest := in.GetId()
+	if idRequest == "" {
+		return nil, status.New(codes.InvalidArgument, "an id is needed").Err()
+	}
+	id, err := uuid.FromString(idRequest)
+	if err != nil {
+		log.Println(err.Error())
+		return nil, status.New(codes.Internal, "id not valid").Err()
+	}
+
+	// start app context
+	appCtx, err := cluster.NewTenantContextWithGrpcContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// if errors are returned, rollback all transactions
+	defer func() {
+		if err != nil {
+			appCtx.Rollback()
+			return
+		}
+		err = appCtx.Commit()
+	}()
+
+	// Fetch the service account
+	serviceAccount, err := cluster.GetServiceAccountByID(appCtx, &id)
+	if err != nil {
+		log.Println(err.Error())
+		return nil, status.New(codes.NotFound, "Service Account Not Found").Err()
+	}
+	// Only update minio user status if the enabled status changed on the update
+	updateMinioUser := !serviceAccount.Enabled
+	if updateMinioUser {
+		err = cluster.SetMinioServiceAccountStatus(appCtx, serviceAccount, true)
+		if err != nil {
+			log.Println(err.Error())
+			return nil, status.New(codes.Internal, "Error Updating Status").Err()
+		}
+	}
+
+	return &pb.ServiceAccount{
+		Id:        serviceAccount.ID.String(),
+		Name:      serviceAccount.Name,
+		AccessKey: serviceAccount.AccessKey,
+		Enabled:   serviceAccount.Enabled,
+	}, nil
+}
+
+func (s *server) DisableServiceAccount(ctx context.Context, in *pb.ServiceAccountActionRequest) (res *pb.ServiceAccount, err error) {
+	idRequest := in.GetId()
+	if idRequest == "" {
+		return nil, status.New(codes.InvalidArgument, "an id is needed").Err()
+	}
+	id, err := uuid.FromString(idRequest)
+	if err != nil {
+		log.Println(err.Error())
+		return nil, status.New(codes.Internal, "id not valid").Err()
+	}
+
+	// start app context
+	appCtx, err := cluster.NewTenantContextWithGrpcContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// if errors are returned, rollback all transactions
+	defer func() {
+		if err != nil {
+			appCtx.Rollback()
+			return
+		}
+		err = appCtx.Commit()
+	}()
+
+	// Fetch the service account
+	serviceAccount, err := cluster.GetServiceAccountByID(appCtx, &id)
```

Comment:
Same as above

Reply:
is already really abstracted, you can use it in the same way from cli and then using `SetMinioServiceAccountStatus`